### PR TITLE
OCPCLOUD-439 - Make "recover from deleted worker machines" test more disruptive

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
Ensure e2e test check recovery from all machines deletion

### CODE REVIEW

File: pkg/framework/machines.go [hunk 1]
```diff
@@ -104,11 +104,13 @@ func GetMachineFromNode(client runtimeclient.Client, node *corev1.Node) (*mapiv1
 }
 
 // DeleteMachine deletes a specific machine and returns an error otherwise
-func DeleteMachine(client runtimeclient.Client, machine *mapiv1beta1.Machine) error {
+func DeleteMachine(client runtimeclient.Client, machines ...*mapiv1beta1.Machine) error {
```
Comment (alexander-demichev) [inline pkg/framework/machines.go:107]: should this be `DeleteMachines` ?

Comment (Danil-Grigorev) [reply pkg/framework/machines.go:107]: No

Comment (arapov) [reply pkg/framework/machines.go:107]: Out of curiosity, how is it not? It gets a number of arguments?

File: pkg/framework/machines.go [hunk 2]
```diff
@@ -104,11 +104,13 @@ func GetMachineFromNode(client runtimeclient.Client, node *corev1.Node) (*mapiv1
...
 	return wait.PollImmediate(1*time.Second, time.Minute, func() (bool, error) {
-		if err := client.Delete(context.TODO(), machine); err != nil {
-			glog.Errorf("Error querying api for machine object %q: %v, retrying...", machine.Name, err)
-			return false, err
+		for _, machine := range machines {
```
Comment (arapov) [inline pkg/framework/machines.go:109]: Iterates here.

File: pkg/framework/machines.go [hunk 3]
```diff
@@ -104,11 +104,13 @@ func GetMachineFromNode(client runtimeclient.Client, node *corev1.Node) (*mapiv1
...
+			if err := client.Delete(context.TODO(), machine); err != nil {
```
Comment (arapov) [inline pkg/framework/machines.go:110]: And delete one by one here....

File: pkg/framework/machines.go (hunk 1 above)
Comment (Danil-Grigorev) [reply pkg/framework/machines.go:107]: It revolves out of the fact you could pass a single instance. There is no reason for referencing it with DeleteMachine`s`

Comment (alexander-demichev) [reply pkg/framework/machines.go:107]: If you are able to pass multiple machines, then it should be called `DeleteMachines`. If you want very much to have a function called `DeleteMachine` make a wrapper.

Comment (arapov) [reply pkg/framework/machines.go:107]: I don't know... This also means that the comment to the function should be clarified. I would be stuck seeing one thing in a comment and other in a function.

File: pkg/framework/machines.go [hunk 4]
```diff
@@ -118,13 +120,16 @@ func DeleteMachine(client runtimeclient.Client, machine *mapiv1beta1.Machine) er
 func WaitForMachineDelete(c client.Client, machines ...*mapiv1beta1.Machine) {
```
Comment (JoelSpeed) [inline pkg/framework/machines.go:120]: If this is now iterating, perhaps we should call it `WaitForMachinesDeleted`?

Comment (enxebre) [reply pkg/framework/machines.go:120]: same for `DeleteMachine`

File: pkg/framework/machines.go [hunk 5]
```diff
@@ -103,28 +103,34 @@ func GetMachineFromNode(client runtimeclient.Client, node *corev1.Node) (*mapiv1
 	return machine, nil
 }
 
-// DeleteMachine deletes a specific machine and returns an error otherwise
-func DeleteMachine(client runtimeclient.Client, machine *mapiv1beta1.Machine) error {
+// DeleteMachines deletes a specific machine and returns an error otherwise
```
Comment (alexander-demichev) [inline pkg/framework/machines.go:106]:
```suggestion
// DeleteMachines deletes a specific machines and returns an error otherwise
```

File: pkg/framework/machines.go [hunk 6]
```diff
@@ -103,28 +103,34 @@ func GetMachineFromNode(client runtimeclient.Client, node *corev1.Node) (*mapiv1
 	return machine, nil
 }
 
-// DeleteMachine deletes a specific machine and returns an error otherwise
-func DeleteMachine(client runtimeclient.Client, machine *mapiv1beta1.Machine) error {
+// DeleteMachines deletes a specific machines and returns an error otherwise
```
Comment (elmiko) [inline pkg/framework/machines.go:106]: just a nit about the comment,

```suggestion
// DeleteMachines deletes the specified machines and returns an error on failure.
```

File: pkg/framework/machines.go [hunk 7]
```diff
@@ -103,28 +103,34 @@ func GetMachineFromNode(client runtimeclient.Client, node *corev1.Node) (*mapiv1
...
 	})
 }
 
-// WaitForMachineDelete polls until the given Machines are not found.
-func WaitForMachineDelete(c client.Client, machines ...*mapiv1beta1.Machine) {
+// WaitForMachinesDeleted polls until the given Machines are not found.
+func WaitForMachinesDeleted(c client.Client, machines ...*mapiv1beta1.Machine) {
 	Eventually(func() bool {
 		for _, m := range machines {
+			currentMachine := &mapiv1beta1.Machine{}
```
Comment (enxebre) [inline pkg/framework/machines.go:123]: why this change?

File: pkg/framework/machines.go [hunk 8]
```diff
@@ -103,28 +103,34 @@ func GetMachineFromNode(client runtimeclient.Client, node *corev1.Node) (*mapiv1
...
 			err := c.Get(context.Background(), client.ObjectKey{
 				Name:      m.GetName(),
 				Namespace: m.GetNamespace(),
-			}, &mapiv1beta1.Machine{})
-
-			if !apierrors.IsNotFound(err) {
-				return false // Not deleted, or other error.
+			}, currentMachine)
+			if apierrors.IsNotFound(err) {
```
Comment (enxebre) [inline pkg/framework/machines.go:128]: why this change?

Comment (Danil-Grigorev) [reply pkg/framework/machines.go:128]: For no reason. Somehow forgot that a name for the machine is randomly generated by a `machineSet`. Will revert the change.

**Comment by Danil-Grigorev** [comment]
@elmiko @JoelSpeed @enxebre @alexander-demichev  Are you guys good with the changes?

**Comment by enxebre** [comment]
thanks!
/retest
/lgtm
